z3c/dependencychecker/package.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+from z3c.dependencychecker.dotted_name import DottedName
+from z3c.dependencychecker.utils import change_dir
+import glob
+import logging
+import os
+import pkg_resources
+import sys
+
+
+logger = logging.getLogger(__name__)
+
+
+class PackageMetadata(object):
+    """Information related to a python source distribution
+
+    It relies heavily on setuptools and pkg_resources APIs.
+    """
+
+    def __init__(self, path):
+        self.distribution_root = self._get_distribution_root(path)
+        self.setup_py_path = self._get_setup_py_path()
+        self.package_dir = self._get_package_dir()
+        self.egg_info_dir = self._get_egg_info_dir()
+        self.name = self._get_package_name()
+        self._working_set = self._generate_working_set_with_package()
+        self.top_level = self._get_sources_top_level()
+
+    @staticmethod
+    def _get_distribution_root(path):
+        if 'setup.py' not in os.listdir(path):
+            logger.error(
+                'setup.py was not found in %s. '
+                'Without it dependencychecker can not work.',
+                path,
+            )
+            sys.exit(1)
+        return path
+
+    def _get_setup_py_path(self):
+        return os.path.join(
+            self.distribution_root,
+            'setup.py',
+        )
+
+    def _get_package_dir(self):
+        """Check where the .egg-info is located"""
+        try_paths = (
+            self.distribution_root,
+            os.path.join(self.distribution_root, 'src', )
+        )
+        for path in try_paths:
+            folder_found = self._check_egg_in_folder(path)
+            if folder_found:
+                logger.debug(
+                    '.egg-info folder found at %s folder',
+                    path,
+                )
+                return path
+
+        logger.error('.egg-info folder could not be found')
+        sys.exit(1)
+
+    @staticmethod
+    def _check_egg_in_folder(path):
```

Comment:
I'd call it `_find_egg_info_in_folder()`. You're not checking anything, you're searching/finding. And "egg_info" instead of "egg" is a tad more explicit.

Reply:
👍 thanks, naming *is hard* as we all know. I tried to be as consistent as possible, but there's always room for improvement